### ISSUE TITLE
[RFC]: PHPDoc change in Laminas\Form\Element\Select::setEmptyOption

### ISSUE THREAD

**Issue by dude920228** [laminas/laminas-form#103]
## RFC

| Q                   | A      |
| ------------------- | ------ |
| Proposed Version(s) | 2.19.0  |
| BC Break?           | No |

### Goal

Allow users to set `Laminas\Form\Element\Select`  empty option as array with HTML attributes without static analysers failing.
### Background
One of my projects required the select fields empty option to be selected and disabled. I found a solution: you can simply pass an array, like this:
```
$this->setEmptyOption([
    'label' => _('Alert types'),
    'attributes' => [
        'selected' => 1,
        'disabled' => 1
    ]
);
```
This solution works, but the PHPDoc at `setEmptyOption` requires either `string` or `null` as a parameter, phpstan fails and so does the pipeline in the project repository.

### Considerations
This should be a minor PHPDoc change, nothing that would break current behavior. The doc should only be changed from this:
`@param  string|null $emptyOption` to this: `@param  string|null|array $emptyOption`

### Proposal(s)

User had the opportunity to do the above trick before, I would only like to make it "official". Users with similar needs would not have to worry about the following phpstan message: 
`Parameter #1 $emptyOption of method Laminas\Form\Element\Select::setEmptyOption() expects string|null, array<string, array<string, int>|string> given.`

**Comment by froschdesign** [comment]
@dude920228 
This is a bug, because the associated class properties have a completely different type:

https://github.com/laminas/laminas-form/blob/90eeb9f3960be594eda3b867183eb51b541dfc4d/src/Element/Select.php#L44-L49

And your example must be added to the documentation:

> ```php
> $this->setEmptyOption([
>     'label' => _('Alert types'),
>     'attributes' => [
>         'selected' => 1,
>         'disabled' => 1
>     ]
> );
> ```

https://docs.laminas.dev/laminas-form/element/select/

**Comment by froschdesign** [comment]
Fixed with #133
